src/view/components/Drawer/DrawerContents/index.tsx

```diff
@@ -25,10 +33,27 @@ const DrawerContents: React.StatelessComponent<OwnProps> = (props) => {
   );
 };
 
-function getDrawerItems(props: OwnProps, routes: RouteInfo[]): JSX.Element[] {
-  return routes
-    .filter(route => route.inDrawer)
-    .map(route => <DrawerItem icon={route.icon} key={route.name} name={route.name} onPress={props.onPress} />);
+function getDrawerItems(props: OwnPropsType, routesInfo: RouteInfo[]): JSX.Element[] {
+  return routesInfo
+    .filter(isDrawerItem)
+    .map(routeInfo => createDrawerItem(routeInfo, props.onPress));
+}
+
+function createDrawerItem(routeInfo: DrawerRouteInfo, onPress: OnPress) {
+  return (
+    <DrawerItem
+      icon={routeInfo.drawerInfo.icon}
+      key={routeInfo.route}
+      name={routeInfo.name}
+      route={routeInfo.route}
+      onPress={onPress}
+    />
+  );
+}
+
+/** Typeguard for routeInfo */
+function isDrawerItem(routeInfo: RouteInfo): routeInfo is DrawerRouteInfo {
+  return !!routeInfo.drawerInfo;
 }
```

Comment:
@beaesguerra @jeinglis @enochtsang @mfalthaw @nscassar 
Just wanted to show this because it's kind of typescript specific, and can be helpful if you're struggling with types. This is a function that returns true or false, and you can see on the return type that it determines whether or not the returned object is of a specific type. So if this function returns true, you're telling the compiler the object is of the type `DrawerRouteInfo`, otherwise it's not.